src/visualizer/svg.js

```diff
@@ -0,0 +1,201 @@
+     * @param {number} y1 the y position of the start of the line
+     * @param {number} x2 the x position of the end of the line
+     * @param {number} y2 the y position of the end of the line, y1 by default
+     * @param {number} width the width of the line, default is determined by the browser size
+     * @returns {SVGLineElement} A line with the coordinates and with provided as arguments
+     */
+    function createLine(x1, y1, x2, y2 = y1, width = svgLineWidth){
+        const line = document.createElementNS("http://www.w3.org/2000/svg", "line");
+        line.setAttribute("x1", x1);
+        line.setAttribute("y1", y1);
+        line.setAttribute("x2", x2);
+        line.setAttribute("y2", y2);
+        line.setAttribute("stroke-width", width);
+        line.setAttribute("style", "stroke: black;");
+        return line;
+    }
+
+    /**
+     * @summary Creates and return a svg cicle
+     * @param {number} cx the x position of the center of the circle
+     * @param {number} cy the y position of the center of the circle
+     * @param {color} color the color of the circle
+     * @param {number} r the radius of the circle
+     * @returns {SVGCircleElement} SVGCicleElement with the color and radius provided as arguments
+     */
+    function createCircle(cx, cy, color, r){
+        const cir = document.createElementNS("http://www.w3.org/2000/svg", "circle");
+        cir.setAttribute("cx", cx);
+        cir.setAttribute("cy", cy);
+        cir.setAttribute("r", r);
+        cir.setAttribute("fill", color);
+        return cir;
+    }
+
+    /**
+     * @summary Creates and returns a svg text element
+     * @param {number} x the x position of the center of the text
+     * @param {number} y the y position of the center of the text
+     * @param {string} text the text to be displayed
+     * @param {string} fontSize the text to be displayed, default is determinded by svgLineSpace
+     * @returns {SVGTextElement} SVGTextElement with the text given as argument
+     */
+    function createText(x, y, text, fontSize = svgTextSize){
+        const textSvg = document.createElementNS("http://www.w3.org/2000/svg", "text");
+        textSvg.setAttribute("x", x);
+        textSvg.setAttribute("y", y);
+        textSvg.textContent = text;
+        textSvg.setAttribute("font-size", fontSize);
+        return textSvg;
+    }
+    // --------------------------------------------------- End of SVG constructors ----------------------------
+    // --------------------------------------------------- Start of math functions only used for svg ----------
+    /**
+     * @summary calculates the x coordinate of the circle's center from the LearningStyleValue(from -11 to 11) to a x coordinate on the line
+     * @param {number} learnStyleNameValue the LearningStyleValue form which to calculate the x value of the circle
+     * @returns {number} Returns the x coordinate of the circle
+     */
+    function circleXValue(learnStyleNameValue){
```

Comment:
Hvis variablerne heri virkelig er nødvendig så gør dem i det mindste const

Reply:
Variablerne er for at gøre det mere tydeligt hvad der sker

Reply:
Er gjort const nu